docs/maven-tutorial.md

```diff
@@ -0,0 +1,251 @@
+tutorial will step you through integrating a Nexus repository into your project.
+
+| Content: |
+|:----------------------------|
+|[Prerequisites](#prerequisites)|
+|[Setting up Nexus](#setting-up-nexus)|
+|[Connecting to Nexus](#connecting-to-nexus)|
+|[See Also](#see-also)|
+
+## Prerequisites
+
+This tutorial assumes that you'll be working with a project that is already set
+up for use with Maven. If you're interested in using Maven with your Java
+project, we highly recommend looking at
+[their guide](https://maven.apache.org/guides/getting-started/index.html).
+
+In addition, be sure to check your application's image for Maven mirror
+capabilities. Many images that use Maven have a 'MAVEN_MIRROR_URL' environment
+variable that you can use to simplify this process.
+
+## Setting up Nexus
+
+Download the official Nexus Docker image and use `new-app` to deploy it.
+```bash
+oc new-app sonatype/nexus
+```
+
+After the build has finished, run:
+```bash
+oc get svc
+```
+
+The output should resemble:
+```bash
+NAME      CLUSTER-IP       EXTERNAL-IP   PORT(S)    AGE
+nexus     172.30.170.249   <none>        8081/TCP   6s
+```
+
+Confirm that Nexus is running by navigating your browser to the new Cluster-IP
+at port 8081. The server may take a bit of time to start, be patient!
+
+> NOTE: Nexus comes pre-configured for the Central Repository, but you may need
+> others for your application. At this point, add any special repositories to
+> Nexus.
+
+Now we'll create a Persistent Volume for Nexus, so that our dependencies will
+not be lost when the Pod running the server terminates. If you do not have admin
+access on your cluster, ask your system administrator to create a Read/Write
+Persistent Volume for you, and skip to the section on creating Persistent Volume
+Claim.
+
+> NOTE: If you don't want persistent storage, continue to
+> [Connecting to Nexus](#connecting-to-nexus). However, your cached
+> dependencies will be lost with the pod.
+
+Create a directory on your host machine that will be writable by OpenShift and
+persistent, in case your host machine reboots.
+```bash
+mkdir nexus-pv
+chmod 777 nexus-pv
+```
+
+Create a file called `pv.yaml` with the following content:
+```yaml
+---
+apiVersion: v1
+kind: PersistentVolume
+metadata:
+  name: nexus-pv
+  labels:
+    type: local
+spec:
+  hostPath:
+    path: "/path/to/your/nexus-pv"
+  accessModes:
+  - ReadWriteOnce
+  capacity:
+    storage: 1Gi
+  persistentVolumeReclaimPolicy: Retain
+```
+
+Import that file into OpenShift. This may require admin permission.
+```bash
+oc create -f pv.yaml
+```
+
+Add a Persistent Volume Claim to the Nexus Deployment Config.
+```bash
+oc volumes dc/nexus --remove --confirm
+oc volumes dc/nexus --add \
+	--name 'nexus-pvc' \
+	--type 'pvc' \
+	--mount-path '/sonatype-work/' \
+	--claim-name 'nexus-pv' \
+	--claim-size '1G'
+```
+
+This removes the previous volume for the Deployment Config, and adds a claim
```

Comment:
@oatmealraisin is correct, but perhaps it would clearer to state it as:
"This removes the previous ephemeral volume for the Deployment Config and adds a claim..."